compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/FirArgumentsToParametersMapper.kt

```diff
@@ -165,8 +165,8 @@ private class FirCallArgumentsProcessor(
     }
 
     private fun processNamedArgument(argument: FirExpression, name: Name) {
-        if (!function.hasStableParameterNames) {
-            addDiagnostic(NamedArgumentNotAllowed(argument, function))
+        function.asBadForNamedArgumentTarget?.let {
```

Comment:
The value doesn't change for each argument, so you can make this a lazy property in the `FirCallArgumentsProcessor` class, e.g.:
```kt
val badForNamedArgumentTarget: BadForNamedArgumentTarget? by lazy = function.asBadForNamedArgumentTarget
```